### PR TITLE
codeintel: Requeue upload when repository is cloning

### PR DESCRIPTION
**Pull request by efritz**
When a repository is currently cloning, requeue the upload record and try again later. Closes https://github.com/sourcegraph/sourcegraph/issues/9700

### CODE REVIEW

File: enterprise/cmd/precise-code-intel-worker/main.go [hunk 1]
```diff
@@ -91,10 +92,9 @@ func mustInitializeStore() store.Store {
 		}
 	})
 
-	store, err := store.New(postgresDSN)
-	if err != nil {
-		log.Fatalf("failed to initialize store: %s", err)
+	if err := dbconn.ConnectToDB(postgresDSN); err != nil {
```
Comment (efritz) [inline enterprise/cmd/precise-code-intel-worker/main.go:95]: I need to get a repository by ID in order to cal ResolveRev. This requires I populate the global db conn.

Until these utils can be refactored into an internal package, is it ok to add this process to the dbutil import allow list?

cc @slimsag

File: dev/check/go-dbconn-import.sh [hunk 2]
```diff
@@ -7,7 +7,7 @@ echo "--- go dbconn import"
 
 set -euf -o pipefail
 
-allowed='^github.com/sourcegraph/sourcegraph/cmd/frontend|github.com/sourcegraph/sourcegraph/enterprise/cmd/frontend|github.com/sourcegraph/sourcegraph/enterprise/cmd/repo-updater'
+allowed='^github.com/sourcegraph/sourcegraph/cmd/frontend|github.com/sourcegraph/sourcegraph/enterprise/cmd/frontend|github.com/sourcegraph/sourcegraph/enterprise/cmd/repo-updater|github.com/sourcegraph/sourcegraph/enterprise/cmd/precise-code-intel-worker'
```
Comment (efritz) [inline dev/check/go-dbconn-import.sh:10]: Will respond to this post-merge if there are any issues.

File: enterprise/cmd/precise-code-intel-worker/main.go (hunk 1 above)
Comment (efritz) [reply enterprise/cmd/precise-code-intel-worker/main.go:95]: https://github.com/sourcegraph/sourcegraph/pull/11648#discussion_r446270745

File: dev/check/go-dbconn-import.sh (hunk 2 above)
Comment (slimsag) [reply dev/check/go-dbconn-import.sh:10]: > I need to get a repository by ID in order to cal ResolveRev. This requires I populate the global db conn.
>
> Until these utils can be refactored into an internal package, is it ok to add this process to the dbutil import allow list?

@efritz did precise-code-intel-worker before this already have a direct connection to postgres? If yes, then this is fine - the intent of this check is to make certain that nobody is creating a connection to postgres without properly responding to DSN changes

Comment (efritz) [reply dev/check/go-dbconn-import.sh:10]: Yes it did. We just never used any methods inside `cmd/frontend/db`. We shut down on postgres dns change so we can reconnect.

Comment (slimsag) [reply dev/check/go-dbconn-import.sh:10]: Cool, then this is fine permanently. Only point of this test is to ensure we do not require admins to configure `POSTGRES_*` env vars in a new location (i.e. that services respect it coming from the config propagation)

**Pull request merged**